### PR TITLE
RHBPMS-4761 - Transaction timeout when running EJB Timers on WAS

### PR DESCRIPTION
**Pull request by mswiderski**
Added local cache for ejb timer service to improve performance after restart
Marked new timers as new to skip the timer service check by name to improve performance

### CODE REVIEW

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EjbSchedulerService.java [hunk 1]
```diff
@@ -53,11 +53,15 @@ public JobHandle scheduleJob(Job job, JobContext ctx, Trigger trigger) {
 		String jobName = getJobName(ctx, id);
 		EjbGlobalJobHandle jobHandle = new EjbGlobalJobHandle(id, jobName, ((GlobalTimerService) globalTimerService).getTimerServiceId());
 		
-		TimerJobInstance jobInstance = scheduler.getTimerByName(jobName);
-		if (jobInstance != null) {
-			return jobInstance.getJobHandle();
+		TimerJobInstance jobInstance = null;
+		// check if given timer job is marked as new timer meaning it was never scheduled before, 
+		// if so skip the check by timer name as it has no way to exist
+		if (!isNewTimer(ctx)) {
```
Comment (tsurdilo) [inline jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EjbSchedulerService.java]: formatting of this seems off

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java [hunk 2]
```diff
@@ -124,24 +132,32 @@ public boolean removeJob(JobHandle jobHandle) {
...
     			Serializable info = timer.getInfo();
     			if (info instanceof EjbTimerJob) {
     				EjbTimerJob job = (EjbTimerJob) info;
     				
     				EjbGlobalJobHandle handle = (EjbGlobalJobHandle) job.getTimerJobInstance().getJobHandle();
+    				localCache.putIfAbsent(jobName, handle.getTimerJobInstance());
     				if (handle.getUuid().equals(jobName)) {
     					logger.debug("Job  {} does match timer and is going to be returned", jobName);
-    					return handle.getTimerJobInstance();
+    					found = handle.getTimerJobInstance();
```
Comment (tsurdilo) [inline jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java]: should we break if found? also formatting seems off here to me

Comment (mswiderski) [inline jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java]: Nope, the point here is to go over all items only once and that's why it does not break directly

File: jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EjbSchedulerService.java (hunk 1 above)
Comment (mswiderski) [inline jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EjbSchedulerService.java]: Not sure, will double check

**Pull request merged**